src/Pages/Login/Login.js

```diff
@@ -1,2 +1,332 @@
-import React from "react";
-import ReactDOM from "react-dom";
+import React, { Component } from "react";
+import { Link } from "react-router-dom";
+import { loginImagesArr } from "./loginImagesData";
+import styled from "styled-components";
+import SocialPlatforms from "./SocialPlatforms";
+
+const loginImgWidth = window.innerWidth;
+
+class Login extends Component {
+  constructor() {
+    super();
+    this.state = {
+      loginImages: [],
+      loginImgCount: 0,
+      currentImgId: 1,
+      loginImgsXcoordinate: 0,
+      loginImgsTransitionDelay: 800,
+      slideIntervalDelay: 5000,
+      userCell: "",
+      userPw: "",
+      socialList: [],
+    };
+  }
+
+  componentDidMount = () => {
+    const { slideIntervalDelay } = this.state;
+    this.setState({
+      loginImages: loginImagesArr,
+      loginImgCount: loginImagesArr.length,
+    });
+    this.timerId = setInterval(this.loginSlideAuto, slideIntervalDelay);
+    window.addEventListener("resize", this.windowReload);
+  };
+
+  componentDidUpdate = (prevProps, prevState) => {
+    const { currentImgId, slideIntervalDelay } = this.state;
+    if (currentImgId !== prevState.currentImgId) {
+      clearInterval(this.timerId);
+      this.timerId = setInterval(this.loginSlideAuto, slideIntervalDelay);
+    }
+  };
+
+  componentWillUnmount = () => {
+    clearInterval(this.timerId);
+  };
+
+  windowReload = () => {
+    window.location.reload();
+  };
+
+  loginSlideAuto = () => {
+    const { loginImages, loginImgsXcoordinate, currentImgId } = this.state;
+    const xIncrement = loginImgWidth * -1;
+    const isLastImg = currentImgId === loginImages.length;
+    this.setState({
+      currentImgId: isLastImg ? 1 : currentImgId + 1,
+      loginImgsXcoordinate: isLastImg ? 0 : loginImgsXcoordinate + xIncrement,
+      loginImgsTransitionDelay: isLastImg ? 0 : 800,
+      slideIntervalDelay: isLastImg ? 0 : 8000,
+    });
+  };
+
+  handleIdPw = (e) => {
+    const { name, value } = e.target;
+    this.setState({
+      [name]: name === "userCell" ? e.target.value.replace(/^\D+$/, "") : value,
+    });
+  };
+
+  validateAndGoToToday = () => {
+    const { userCell, userPw } = this.state;
+    if (userCell.length >= 10 && userPw.length >= 5) {
+      this.props.history.push("/today");
+    } else {
+      alert("휴대폰 번호 10-11자, 비밀번호 5자 이상 입력해주세요.");
+    }
+  };
+
+  validateAndGoToTodayOnEnter = (e) => {
+    if (e.key === "Enter") {
+      this.validateAndGoToToday();
+    }
+  };
+
+  render() {
+    const {
+      loginImages,
+      loginImgsXcoordinate,
+      loginImgsTransitionDelay,
+      userCell,
+      userPw,
+    } = this.state;
+    return (
+      <StyledLogin className="Login">
+        <BackgroundImgContainerUl
+          className="backgroundImagesContainer"
+          xCor={loginImgsXcoordinate}
+          transDelay={loginImgsTransitionDelay}
+        >
+          {!!loginImages.length &&
+            loginImages.map((image) => {
```

Comment:
```suggestion
          {loginImages?.map((image) => {
```
옵셔널 체이닝 사용해볼 수도 있겠습니다

Reply:
네!